source/opencascade/utilities.cc

```diff
@@ -267,6 +276,64 @@ namespace OpenCASCADE
     AssertThrow(OK, ExcMessage("Failed to write IGES file."));
   }
 
+
+  TopoDS_Shape
+  read_STL(const std::string &filename)
+  {
+    StlAPI_Reader reader;
+    TopoDS_Shape  shape;
+    reader.Read(shape, filename.c_str());
+    return shape;
+  }
+
+  void
+  write_STL(const TopoDS_Shape &shape,
+            const std::string & filename,
+            const double        deflection,
+            const double        angular_deflection,
+            const bool          sew_different_faces)
+  {
+    bool mesh_is_present = true;
+    Handle(StlMesh_Mesh) stl_mesh;
+    TopoDS_Shape               shape_to_be_written = shape;
+    TopLoc_Location            Loc;
+    std::vector<TopoDS_Vertex> vertices;
+    std::vector<TopoDS_Edge>   edges;
+    std::vector<TopoDS_Face>   faces;
+    OpenCASCADE::extract_geometrical_shapes(shape, faces, edges, vertices);
+    for (unsigned int i = 0; i < faces.size(); ++i)
+      {
+        Handle(Poly_Triangulation) theTriangulation =
+          BRep_Tool::Triangulation(faces[i], Loc);
+        if (theTriangulation.IsNull())
+          mesh_is_present = false;
+      }
+    if (!mesh_is_present)
+      {
+        if (sew_different_faces)
+          {
+            const double          tolerance = 1e-06;
```

Comment:
Can you make this tolerance also an argument of the function (say `sewer_tolerance`)

Reply:
yep, I will recompile to check that everything works fine. Then I will rebase and push

Reply:
done